src/xdocs/config_javadoc.xml

```diff
@@ -1281,6 +1288,16 @@ public boolean isSomething()
 &lt;module name="JavadocType"&gt;
   &lt;property name="scope" value="private"/&gt;
   &lt;property name="excludeScope" value="package"/&gt;
+&lt;/module&gt;
+        </source>
+
+        <p>
+          To allow missing comments for classes annotated with
+          <code>@SpringBootApplication</code> and <code>@Configuration</code>:
+        </p>
+        <source>
+&lt;module name="JavadocType"&gt;
+  &lt;property name="allowedAnnotations" value="SpringBootApplication,Configuration"/&gt;
```

Comment:
I think that my example is good enough. It is actually better than the check `JavadocMethod` (which is what I based my changes on), which doesn't even have an example like mine (and it's already on production) http://checkstyle.sourceforge.net/config_javadoc.html#JavadocMethod

Reply:
>I think that my example is good enough.

Should be enhanced.

>It is actually better than the check JavadocMethod (which is what I based my changes on), which doesn't even have an example like mine (and it's already on production) 

Sorry that we have bunch of bad examples in documentation. All new stuff we demand do match best practices that we know now.

Reply:
I'm just wondering if we're falling for the trap of the better being the enemy of the good. As a checkstyle user, I never had a problem understanding the checkstyle documentation. But maybe that's just me.

I've updated the docs with side by side example of source code and matching checkstyle configuration.

Reply:
I hope you enjoying checkstyle documentation, we improved it a lot during few past year, and work is still in progress...
But I am working not only with a lot of engineers, and  not all of them very experienced, so come to such practices in documentation due a lot of requests to clarify or explain.

I want to make docs to be clear for all. After discussion in issue tracker and classifying issue as won't fix or ... We usually update doc to avoid such questions in future

Reply:
I hope you enjoying checkstyle documentation, we improved it a lot during few past year, and work is still in progress...
But I am working not only with a lot of engineers, and  not all of them very experienced, so come to such practices in documentation due a lot of requests to clarify or explain.

I want to make docs to be clear for all. After discussion in issue tracker and classifying issue as won't fix or ... We usually update doc to avoid such questions in future

Reply:
What I would personally like to see is a collection of popular presets and somehow sorted by the number of downloads. I have created my own set of rules, which I don't like doing. I'd rather follow a popular preset, if it's close to my taste. Something like what ESLint is doing.

The documentation is great. What I think might improve it is if the core team of checkstyle gets a bit more diverse. I'm always trying to make sure the documentation I write is in proper English. I'm not a native English speaker, so I make mistakes, like everybody. When all developers have the same background, it's difficult for them to recognize their own grammar mistakes, so diversity helps there.

Reply:
> What I would personally like to see is a collection of popular presets and somehow sorted by the number of downloads.

if you know how to do this, please create issue and send us PR :) .

> What I think might improve it is if the core team of checkstyle gets a bit more diverse. 

Please be welcome to be part of team :) .

> When all developers have the same background, it's difficult for them to recognize their own grammar mistakes, so diversity helps there

I try attract more and more people to project. There are some web services that can help us with localization - #6191.
Here is how I try to verify translations now - https://github.com/checkstyle/checkstyle/pull/6067#discussion_r217240504
If you know better way - let us know.

Reply:
Thank you! I'll try to contribute something else in the future. I think I'd like to do the code cleanup we discussed that all test methods should start with `test`, which currently isn't the case.

Reply:
Please create and issue on this to discuss details